src/main/java/uk/gov/hmcts/sptribs/systemupdate/schedule/migration/SystemMigrateGlobalSearchTask.java

```diff
@@ -43,31 +51,34 @@ public SystemMigrateGlobalSearchTask(AuthTokenGenerator authTokenGenerator, CcdS
 
     @Override
     public void run() {
-        final User user = idamService.retrieveSystemUpdateUserDetails();
-        final String serviceAuth = authTokenGenerator.generate();
+        if (globalSearchMigrationEnabled) {
+            final User user = idamService.retrieveSystemUpdateUserDetails();
+            final String serviceAuth = authTokenGenerator.generate();
 
-        try {
-            final BoolQueryBuilder query =
-                boolQuery()
-                    .must(boolQuery()
-                        .mustNot(existsQuery("data.SearchCriteria"))
-                    );
-
-            final List<CaseDetails> casesToMigrateSearchCriteria =
-                ccdSearchService.searchForAllCasesWithQuery(query, user, serviceAuth);
-            log.info("Cases:" + casesToMigrateSearchCriteria.size());
-
-            for (final CaseDetails caseDetails : casesToMigrateSearchCriteria) {
-                triggerSystemMigrateSearchCriteriaTask(user, serviceAuth, caseDetails);
-            }
+            try {
+                final BoolQueryBuilder query =
+                    boolQuery()
+                        .must(boolQuery()
+                            .mustNot(existsQuery("data.SearchCriteria"))
+                            .must(matchQuery("reference", Long.parseLong(globalSearchTestCaseReference)))
```

Comment:
Can we avoid including this in the query if there is no case reference mentioned?